module-1/lab-errhand_listcomp/your-code/main.py

```diff
@@ -9,6 +9,7 @@
 #Insert here the module/library import statements 
```

Comment:
Question 4: Really good and simple approach. Another way without subtracting anything:
```Python
my_list = [num for num in range(-10,1)]
print(my_list)
```
Question 7: good approach!!. You can write your code making it more readable. One way: 
```Python
vowels = ['a','e','i','o','u',' ']
non_vowels = [letter for letter in teststring if letter.lower() not in vowels]
print(non_vowels)
```
Question 8: the same as in question 7. Keep code as simple as possible and try to make it readable for others. 
Question 10: Remember to use always relative paths instead of absolute paths. Your solutions is written as a comment, I have tried to execute and I guess this approach doesn't work. This is the example shown in class on Friday: 
```Python
import os
from os import walk
mypath = '../madrid-oct-2018'
files = [folder for folder in next(os.walk(mypath))[1]]
print(files)
```
Question 11: Your code doesn't throw 4 lists of 10 random numbers. Only this chunk of code throws what expected: 
```Python
random = np.random.random((4,10))
```
Question 12: Good approach!!. Another one is using that library you loves, numpy, and its flatten function. 
```Python
#AlbertoG's approach:
import numpy as np
flatten_list = np.array(list_of_lists).flatten()
```
Questions 15 to 18 are not correct. You must review handling exceptions learning unit. One way of doing it: 
```Python
#15. Handle the exception thrown by the code below by using try and except blocks. 
#Then use a finally block to print 'All Done.'
# Check in provided resources the type of error you may use. 

x = 5
y = 0

try:
    z = x/y
except ZeroDivisionError:4
    print ("Can't divide by Zero!")
finally:
    print ('All Done!')


#16. Handle the exception thrown by the code below by using try and except blocks. 
# Check in provided resources the type of error you may use. 

abc=[10,20,20]

#print(abc[3])
try:
    i = int(input("Enter item index to print: "))
except IndexError:
    print ("List index out of range!")
else:
    print(abc[i])


#17. Handle at least two kind of different exceptions when dividing a couple of numbers provided by the user. 
# Hint: take a look on python input function. 
# Check in provided resources the type of error you may use. 

try:
    x = int( input( "enter a number: " ) )
    y = int( input( "enter another number: " ) )
    print( x, '/', y, '=', x/y )
except ZeroDivisionError:
    print( "Can't divide by 0!" )
except ValueError:
    print( "That doesn't look like a number!" )
except:
    print( "something unexpected happend!" )


#18. Handle the exception thrown by the code below by using try and except blocks. 
# Check in provided resources the type of error you may use. 

try:
    f = open('test','r')
    f.write('Test write this')
except IOError:
   print ("Error: File not found or is read-only")
else:
   print ("Content written successfully")
   f.close()


#19. Handle the exceptions that can be thrown by the code below using try and except blocks. 
#Hint: the file could not exist and the data could not be convertable to int

try:
    fp = open('myfile.txt')
    line = f.readline()
    i = int(s.strip())
except (IOError,ValueError) as e:
    print ("Please check the file,either the file is read-only or the data can't be converted to an integer.",e.errno)
except: 
    print ("Unexpected error")
```